Translation Issues on Bahasa App

syedowaisali90 opened the pull request:
Hi @vincent-karuri I have created a PR for review.
Close #642

syedowaisali90 commented:
`v_max_length.err` and `v_min_length.err` translation keys not auto-generated by MLS Asset Generator so I have put these translations manually.

vincent-karuri commented on rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/interactor/CovidPatientVisitFragmentInteractor.java:
```diff
@@ -27,7 +29,7 @@ public CovidPatientVisitFragmentInteractor() {
         List<Visit> visits = new ArrayList<>();
         for (int i = 0; i < eventClients.size(); i++) {
-            visits.add(new Visit(String.format("Visit %d", i + 1),
+            visits.add(new Visit(String.format(RDTApplication.getInstance().getResources().getString(R.string.lbl_visit) + " %d", i + 1),
```
Could you move the `%d` to the string resource?

vincent-karuri commented on rdt-open-reader/src/covidIndonesia/res/values-in/strings.xml:
```diff
@@ -28,4 +28,7 @@
         <item>en</item>
         <item>in</item>
     </string-array>
+    <string name="lbl_back">Back</string>
+    <string name="lbl_gender_id">Female . ID: 99358357</string>
+    <string name="lbl_visit">Visit</string>
```
Any reason to add these? They're not translated.

syedowaisali90 replied:
removed.

vincent-karuri commented on rdt-open-reader/src/main/res/values/strings.xml:
```diff
@@ -53,5 +53,8 @@ Please click on the <i>Locate Me</i> button at the bottom of the screen and we w
     <string name="rdt_not_supported">Sorry, the selected RDT is not currently supported.</string>
     <string name="male">Male</string>
     <string name="female">Female</string>
+    <string name="lbl_back">Back</string>
+    <string name="lbl_gender_id">Female . ID: 99358357</string>
+    <string name="lbl_visit">Visit</string>
```
Aren't these duplicates https://github.com/onaio/rdt-standard/pull/648/files#diff-1a659ac4e58d657a59f0333681eb88c15c48f0bb32a9a38b51b9c72de466009dR32?

syedowaisali90 replied:
this command failed if remove these strings
`./gradlew :rdt-open-reader:clean :rdt-open-reader:testMalariaKenyaDebugUnitTestCoverage`

vincent-karuri commented:
There are a couple of unaddressed issues from the PDF attached to the issue. Please look into them.

syedowaisali90 commented:
> There are a couple of unaddressed issues from the PDF attached to the issue. Please look into them.

I have seen 4 screen issues in PDF could you highlight the unaddressed issue.

vincent-karuri commented:
There's a comment on changing the order of questions and most translations have not been added though I think they should be added via Transifex anyway.

vincent-karuri commented on rdt-open-reader/src/covidIndonesia/res/values/strings.xml:
```diff
@@ -29,4 +29,7 @@
         <item>in</item>
     </string-array>
+    <string name="lbl_back">Back</string>
+    <string name="lbl_gender_id">Female . ID: 99358357</string>
```
The string that appears under the patient name is dynamically generated, I don't think this is where it's picked from.

syedowaisali90 replied:
yes, it's just moved here instead of hardcoded typos

vincent-karuri replied:
Ok, is the dynamic one translated?

syedowaisali90 replied:
No, it's not dynamically translated.

vincent-karuri replied:
I think that's one of the issues raised.

syedowaisali90 replied:
so how can we translate the dynamic data? which comes directly from DB.

vincent-karuri replied:
I think we've done this before for the register page?

syedowaisali90 replied:
fixed.

vincent-karuri commented on rdt-open-reader/src/covidIndonesia/assets/json.form/patient-registration-form-original.json:
```diff
@@ -186,7 +186,7 @@
         "expanded": false,
         "duration": {
-          "label": "DoB"
+          "label": "Age"
```
I think this is a dob and not age field.

syedowaisali90 replied:
yes, reverted

syedowaisali90 replied:
Can you just confirm for me the first issue of file:///home/vend/Downloads/Translation.issues_SID.pdf this PDF which is related the to DOB field so In thePDF says "Tanggal lahir" to "Usia" so for this reason I have changed the "DoB" to "Age" so could you suggest me the correct fix of this issue

vincent-karuri replied:
The change in Bahasa shouldn't affect the English version.
